### ISSUE TITLE
Plugin not working anymore

### ISSUE THREAD

**Issue by Invictaz** [BuXXe/movian-bs-plugin#5]
Says with every video: file not available.
The website with various sources still works.
Maybe because of new Captcha on top of the video's?

For example:

https://bs.to/serie/Knight-Rider/1/1-Michael-Knight-Teil-1/OpenLoad

**Comment by BuXXe** [comment]
i will have a look

**Comment by LFriede** [comment]
OpenLoad is broken, i mostly use Vivo, it works tested it just now. Check if your system time is correct, bs.to API requires a valid system time.

**Comment by Invictaz** [comment]
![sidesync_capture_20180304220137_1](https://user-images.githubusercontent.com/20338545/36950541-a2cd78d4-1ff7-11e8-9dc2-917622cf7768.jpg)

That did not work in the Chrome version so I reported https://movian.tv/issues/3376, and the browse function does not work, see screenshot.

I tested both on Android and on Chrome versions of Movian.

**Comment by LFriede** [comment]
You can go to the plugins settings and activate "online timesync" this will not fix the browse function but may be a workaround for the "file not available" message.

**Comment by Invictaz** [comment]
@LFriede That did fix the "Browse" function!

Unfortunately not the episodes. All still say: file not available

![sidesync_capture_20180304231037_1](https://user-images.githubusercontent.com/20338545/36951217-46bbe54e-2001-11e8-84d3-a4fc6d1b240f.jpg)

**Comment by BuXXe** [comment]
I just tested it on my setup and cannot find any problem / recreate your issue. I can browse without problems and even play episodes.

**Comment by Invictaz** [comment]
@BuXXe here is my logfile. Even with disabled HTTP connection reuse it does not work.
Running on Android version 5.0.491

[movian-0.log](https://github.com/BuXXe/movian-bs-plugin/files/1779120/movian-0.log)

**Comment by Invictaz** [comment]
@BuXXe @LFriede Here is the log from my PS3 running on 5.0.504 (which is not even publically released)

[movian-1.log](https://github.com/BuXXe/movian-bs-plugin/files/1779128/movian-1.log)

**Comment by BuXXe** [comment]
the logs dont seem to give much more info. [DEBUG]:cannot read property 1 of null and  [DEBUG]:{} on the PS3 log
after a successful request it seems like the request body does not fit the expectations but cant say why.

**Comment by Invictaz** [comment]
If I can log more I will but I don't know how.

**Comment by BuXXe** [comment]
i will write some debug log to the plugin and give it to you in a sec

**Comment by BuXXe** [comment]
[bs.to.zip](https://github.com/BuXXe/movian-bs-plugin/files/1779162/bs.to.zip)

this one should have debug set to true for vivo links

**Comment by Invictaz** [comment]
@BuXXe New log 

Error is 0x80010224 errno=36
[movian-2.log](https://github.com/BuXXe/movian-bs-plugin/files/1779181/movian-2.log)

Problem is also discussed here, and the fix is here also suggested by @lprot. Maybe you havn't seen it yet: 

https://movian.tv/issues/3262

**Comment by BuXXe** [comment]
?? this does not seem to be the issue here cause there is no problem with ddos guard redirections

**Comment by Invictaz** [comment]
Yes but the error 0x800110224 is the same. I saw it in his log too.

**Comment by BuXXe** [comment]
the error happens throughout the log so this is not connected to the plugin but the system itself. I really cannot see why the plugin does not work and i guess it has to do with movian instead of the plugin cause i really cannot reproduce the error.sorry

**Comment by Invictaz** [comment]
Maybe because you have this fix?

https://github.com/andoma/movian/pull/230

**Comment by BuXXe** [comment]
Cannot say much about that, i use the plugin on my ps3 with movian 5.0.487. and ps3 os 4.750

so i think that the "newer" versions of your system / movian might have some problems. otherwise it would be related to your setup

**Comment by Invictaz** [comment]
Here is a new log with 5.0.479

00:00:21.278: HTTP            [DEBUG]:Connecting to bs.to:80
00:00:21.304: NET             [DEBUG]:Connecting fd 0x105 = 0x80010224 errno=36

This happens immediately after. And Bmovies plugin works fine. 

Maybe @lprot knows

[movian-3.log](https://github.com/BuXXe/movian-bs-plugin/files/1779255/movian-3.log)

**Comment by BuXXe** [comment]
each request has this error in the log

**Comment by Invictaz** [comment]
In my log I see also the DDOS-GUARD.

What is strange is that in the Movian issue page the redirection is followed until vivo.sx

And in my situation I don't even reach the vivo.sx page. So I think the redirection was unsuccesfull.

**Comment by Invictaz** [comment]
This error log was made over a 4G connection.
You see it tries to open a NULL URL

00:00:26.593: Navigator       [INFO ]:Tried to open NULL URL
00:00:27.229: Navigator       [INFO ]:Tried to open NULL URL

 [movian-0.txt](https://github.com/BuXXe/movian-bs-plugin/files/1779304/movian-0.txt)

**Comment by BuXXe** [comment]
Perhaps this has to do with UserAgent but this was an error fixed long ago. i dont think that i could fix this issue in my plugin when its rather a movian error. correct me if i am wrong.

**Comment by Invictaz** [comment]
And if you spoof the useragent on PC?

**Comment by BuXXe** [comment]
i dont think that it is realted to this cause then it would already break on bs.to

**Comment by LFriede** [comment]
Yesterday i've seen the problem after installing the Movian app on my phone. First try, same error as on the first screenshot. But i wasn't able to reproduce it later, it just worked on later tries.

> @LFriede That did fix the "Browse" function!

I don't think so because this changes only one request and there should be no influence to the browse function. Maybe it was random.

**Comment by BuXXe** [comment]
this would support the suggestion that it is not related to the plugin but the setup / movian itself

**Comment by Invictaz** [comment]
@LFriede since you forked it now, can you replicate the error or fix it?

**Comment by Invictaz** [comment]
@LFriede I tried your version and it has the same error. I found out why it does not work

If you go to https://bs.to/api/watch/1707936 on pc, you need to solve a Captcha 

![image](https://user-images.githubusercontent.com/20338545/37437079-83c1cb60-27eb-11e8-8de4-a358ac286373.png)

After that you end up on: 

https://www.flashx.ws/embed.php?c=315dc8871b2507aaaf0b484ef9beb95e

In this error log you don't even reach that Flashx url at all 

I see also some other stuff

00:00:29.025: HTTP-22         [DEBUG]:http://bs.to/serie/Knight-Rider/1/1-Michael-Knight-Teil-1/Streamango: Following redirect to https://bs.to/serie/Knight-Rider/1/1-Michael-Knight-Teil-1/Streamango, (premanent)
00:00:29.025: HTTP-22         [DEBUG]:0x000000: 3c 68 74 6d 6c 3e 0d 0a  3c 68 65 61 64 3e 3c 74    <html>..<head><t
00:00:29.025: HTTP-22         [DEBUG]:0x000010: 69 74 6c 65 3e 33 30 31  20 4d 6f 76 65 64 20 50    itle>301 Moved P
00:00:29.025: HTTP-22         [DEBUG]:0x000020: 65 72 6d 61 6e 65 6e 74  6c 79 3c 2f 74 69 74 6c    ermanently</titl
00:00:29.025: HTTP-22         [DEBUG]:0x000030: 65 3e 3c 2f 68 65 61 64  3e 0d 0a 3c 62 6f 64 79    e></head>..<body
00:00:29.026: HTTP-22         [DEBUG]:0x000040: 20 62 67 63 6f 6c 6f 72  3d 22 77 68 69 74 65 22     bgcolor="white"
00:00:29.026: HTTP-22         [DEBUG]:0x000050: 3e 0d 0a 3c 63 65 6e 74  65 72 3e 3c 68 31 3e 33    >..<center><h1>3
00:00:29.026: HTTP-22         [DEBUG]:0x000060: 30 31 20 4d 6f 76 65 64  20 50 65 72 6d 61 6e 65    01 **Moved Permane**
00:00:29.026: HTTP-22         [DEBUG]:0x000070: 6e 74 6c 79 3c 2f 68 31  3e 3c 2f 63 65 6e 74 65    **ntly**</h1></cente
00:00:29.026: HTTP-22         [DEBUG]:0x000080: 72 3e 0d 0a 3c 68 72 3e  3c 63 65 6e 74 65 72 3e    r>..<hr><center>
00:00:29.026: HTTP-22         [DEBUG]:0x000090: 62 73 3c 2f 63 65 6e 74  65 72 3e 0d 0a 3c 2f 62    bs</center>..</b
00:00:29.026: HTTP-22         [DEBUG]:0x0000a0: 6f 64 79 3e 0d 0a 3c 2f  68 74 6d 6c 3e 0d 0a       ody>..</html>..
00:00:29.026: HTTP            [DEBUG]:Parking connection to bs.to:80 (cid=12) (expire in 60s) -- **Location changed**

Here is the logfile, I cleaned it a bit for my UPNP sources and local indexing

[logfile2.txt](https://github.com/BuXXe/movian-bs-plugin/files/1813536/logfile2.txt)

**Comment by LFriede** [comment]
> I tried your version and it has the same error.

My latest patches are doing something completely different. Not related to this error.

> If you go to https://bs.to/api/watch/1707936 on pc, you need to solve a Captcha

The API link should return a json encoded message which says that u are not authorized if you open it in a browser. We are getting the video URL with the API because you don't need a captcha then. The captcha should only appear on the "out/..." URL. The plugin trys the "api/watch/..." first and then falls back to the "out/..." if it gets nothing back from the API, this seems to be the case here.

**Comment by Invictaz** [comment]
Well the Captcha also appears on the /api/watch. Try it for yourself.

**Comment by LFriede** [comment]
I did, and it says `{"error":"unauthorized","unauthorized":"no_token"}`

**Comment by Invictaz** [comment]
And what can we do about that?

**Comment by LFriede** [comment]
I think we should ask one of the bs.to devs.
@BuXXe You have a contact, don't you?

**Comment by BuXXe** [comment]
yes but as long as we cannot reproduce  or at least say what the problem is, i think it would be useless to contact them

**Comment by LFriede** [comment]
> Well the Captcha also appears on the /api/watch. Try it for yourself.

I logged the return text of the api request today and as soon as I try to spoof my location with Tunnelbear VPN I can see this behaviour in the Movian logfile. If I open the URL in my browser I still see "unauthorized" because the token is missing. If I remove the token in Movian I see "unauthorized" with or without VPN.

To me it looks like they check for the correct token first, then they seem to have a Geo/VPN check. Without token you always get the error in JSON format, if you have a token but wrong location (assumption) you see the captcha.

One thing still makes me wonder, Invictaz gets the captcha in his browser, I am not.

**Comment by Invictaz** [comment]
![image](https://user-images.githubusercontent.com/20338545/37685968-c6c30b0a-2c95-11e8-85ba-c9f2e7852d19.png)

Movian 5.0.531 

As you can see sometimes it wants to go to /api/watch and sometimes to /out

New logfile: 

My IP is in The Netherlands.

[logfile3.txt](https://github.com/BuXXe/movian-bs-plugin/files/1831436/logfile3.txt)

**Comment by xNeo92x** [comment]
Hello, are there any news on this or is this project dead?

**Comment by BuXXe** [comment]
The bs.to developers are not wanting to give the API keys / access. Due to the fact that movian cant support the captcha stuff I guess we have to wait if bs.to might give a public api
